src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -921,10 +930,10 @@ impl<'a, 'tcx> Visitor<'tcx> for Checker<'a, 'tcx> {
         self.super_place(place, context, location);
         match *place {
             Place::Base(PlaceBase::Local(_)) => {}
-            Place::Base(PlaceBase::Static(ref global)) => {
-                assert!(global.promoted.is_none());
+            Place::Base(PlaceBase::Static(box Static{ kind: StaticKind::Promoted(_), .. })) => {}
```

Comment:
make this `unreachable!()`, too, like all the other sites in the file. It wasn't before your PR, but it should have been